crates/transaction-pool/src/pool/parked.rs

```diff
@@ -79,10 +86,99 @@ impl<T: ParkedOrd> ParkedPool<T> {
         Some(tx.transaction.into())
     }
 
-    /// Removes the worst transaction from this pool.
-    pub(crate) fn pop_worst(&mut self) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
-        let worst = self.best.iter().next().map(|tx| *tx.transaction.id())?;
-        self.remove_transaction(&worst)
+    /// Get transactions by sender
+    pub(crate) fn get_txs_by_sender(
+        &self,
+        sender: &Address,
+    ) -> Vec<Arc<ValidPoolTransaction<T::Transaction>>> {
+        let mut txs = Vec::new();
+        for tx in self.by_id.values() {
+            if tx.transaction.sender() == *sender {
+                txs.push(tx.transaction.clone().into());
+            }
+        }
```

Comment:
if we convert to betreemap this can be significantly improved

Reply:
New map includes `sender => [TransactionId]` so we don't have to iterate over all transactions any more